app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -929,10 +930,44 @@ def private_load_allowed_networks(vlans, uid_ems_cluster)
       vlans['<Template>'] = _('<Use template nics>')
     end
 
+    def private_load_allowed_networks(vlans, uid_ems_cluster)
+      profiles = local_get_vnic_profiles_in_cluster(uid_ems_cluster)
+      profiles.each do |profile, profile_network|
+        vlans[profile.uid_ems] = "#{profile.name} (#{profile_network.name})"
+      end
+
+      vlans['<Empty>'] = _('<No Profile>')
+      vlans['<Template>'] = _('<Use template nics>')
+    end
+
+    def local_get_vnic_profiles_in_cluster(uid_ems_cluster)
+      cluster_profiles = {}
+      profiles = all_vnic_profiles_as_lans_in_ems(ext_management_system)
+      cluster_networks = all_networks_as_switches_in_cluster(uid_ems_cluster)
+      profiles.each do |p|
+        profile_network = cluster_networks.detect { |n| n.id == p.switch.id }
+        if profile_network
+          cluster_profiles[p] = profile_network
+        end
+      end
+      cluster_profiles
+    end
+
+    def all_networks_as_switches_in_cluster(uid_ems_cluster)
+      cluster = EmsCluster.find_by(:uid_ems => uid_ems_cluster)
+      hosts = cluster.hosts
+      host_switches = hosts.collect(&:host_switches).flatten.compact.uniq
+      host_switches.collect(&:switch).flatten.compact.uniq
```

Comment:
There is a relation defined on ems_cluster to get all of the switches from all of the hosts, should be able to just do `cluster.switches`

Reply:
hmm I will try, I think it did not work for me, but I will try again.